ui/src/app/views/copy-notebook-modal/component.tsx

```diff
@@ -0,0 +1,185 @@
+
+export interface CopyNotebookModalProps {
+  fromWorkspaceNamespace: string;
+  fromWorkspaceName: string;
+  fromNotebook: FileDetail;
+  onClose: Function;
+  onCopy: Function;
+}
+
+export interface CopyNotebookModalState {
+  writeableWorkspaces: Array<Workspace>;
+  destination: Workspace;
+  newName: string;
+  state: State;
+  errorMsg: string;
+  loading: boolean;
+}
+
+const boldStyle = {
+  fontWeight: 700
+};
+
+export class CopyNotebookModal extends React.Component<CopyNotebookModalProps,
+CopyNotebookModalState> {
+  constructor(props: CopyNotebookModalProps) {
+    super(props);
+
+    this.state = {
+      writeableWorkspaces: [],
+      newName: props.fromNotebook.name,
+      destination: null,
+      state: State.FORM,
+      errorMsg: '',
+      loading: true
+    };
+  }
+
+  componentDidMount() {
+    workspacesApi().getWorkspaces()
+      .then((response) => {
+        this.setState({
+          writeableWorkspaces: response.items
+            .filter(item => new WorkspacePermissions(item).canWrite)
+            .map(workspaceResponse => workspaceResponse.workspace),
+          loading: false
+        });
+      });
+  }
+
+  save() {
+    this.setState({ loading: true });
+
+    workspacesApi().copyNotebook(
+      this.props.fromWorkspaceNamespace,
+      this.props.fromWorkspaceName,
+      this.props.fromNotebook.name,
+      {
+        toWorkspaceName: this.state.destination.id,
+        toWorkspaceNamespace: this.state.destination.namespace,
+        newName: this.state.newName
+      }
+      ).then((response) => {
+        this.setState({ state: State.SUCCESS, loading: false });
+        this.props.onCopy(response);
+      }).catch((response) => {
+        const errorMsg = response.status === 400 ?
+          'Notebook with the same name already exists in the targeted workspace.' :
+          'An error occurred while copying. Please try again.';
+
+        this.setState({
+          errorMsg: errorMsg,
+          state: State.ERROR,
+          loading: false
+        });
+      });
+  }
+
+  goToDestinationWorkspace() {
+    window.location.href = this.getWorkspaceUrl(this.state.destination);
+  }
+
+  getWorkspaceUrl(workspace: Workspace) {
+    return ['workspaces', workspace.namespace, workspace.id, 'notebooks'].join('/');
+  }
+
+  render() {
+    return (
+      <Modal onRequestClose={this.props.onClose}>
+        <ModalTitle>Copy to Workspace</ModalTitle>
+          { this.state.loading ?
+            <ModalBody style={{ textAlign: 'center' }}><Spinner /></ModalBody> :
+            <ModalBody>
+                {this.state.state === State.FORM && this.renderFormBody()}
+                {this.state.state === State.ERROR && this.renderErrorBody()}
+                {this.state.state === State.SUCCESS && this.renderSuccessBody()}
+            </ModalBody>
+        }
+        <ModalFooter>
+          <Button type='secondary' onClick={this.props.onClose}>
+              {this.getCloseButtonText()}
```

Comment:
See above on the state variable. I think we can eliminate the `Success` state and then turn this into a ternary operator.